session/xmlrpc.go

```diff
@@ -82,8 +79,8 @@ func (x *XmlRpcTransport) DoRequest(
 		}
 
 		timeout := DefaultXmlRpcTimeout
-		if x.Timeout != 0 {
-			timeout = x.Timeout
+		if sess.Timeout != 0 {
```

Comment:
timeout should never be unlimited. it is a potential problem for both client and server. users can come up with a really long timeout if they wish.